operator/api/redpanda/v1alpha3/redpanda_conversion.go

```diff
@@ -0,0 +1,954 @@
+		return &redpandav1alpha2.RedpandaClusterSpec{
+			Image:         image,
+			Enterprise:    convertTo[*redpandav1alpha2.Enterprise](src).(*redpandav1alpha2.Enterprise),
+			ClusterDomain: nilIfZero(src.Spec.ClusterDomain),
+			Config: &redpandav1alpha2.Config{
+				RPK:  rpkConfig,
+				Node: nodeConfig,
+
+				// All ClusterConfig gets pushed into ExtraClusterConfiguration
+				Tunable:                   nil,
+				Cluster:                   nil,
+				ExtraClusterConfiguration: convertTo[redpandav1alpha2.ClusterConfiguration](src).(redpandav1alpha2.ClusterConfiguration),
+
+				// Unused fields
+				SchemaRegistryClient: nil,
+				PandaProxyClient:     nil,
+			},
+			TLS:         convertTo[*redpandav1alpha2.TLS](src).(*redpandav1alpha2.TLS),
+			Listeners:   convertTo[*redpandav1alpha2.Listeners](src).(*redpandav1alpha2.Listeners),
+			Statefulset: convertTo[*redpandav1alpha2.Statefulset](src).(*redpandav1alpha2.Statefulset),
+			Resources: &redpandav1alpha2.Resources{
+				Limits:   (*map[corev1.ResourceName]resource.Quantity)(&src.Spec.NodePoolSpec.BrokerTemplate.Resources.Limits),
+				Requests: (*map[corev1.ResourceName]resource.Quantity)(&src.Spec.NodePoolSpec.BrokerTemplate.Resources.Requests),
+				CPU:      nil,
+				Memory:   nil,
+			},
+
+			// TODO Audit this list.
+			Affinity:                   nil,
+			AuditLogging:               nil,
+			Auth:                       nil,
+			CommonLabels:               nil,
+			Connectors:                 nil,
+			Console:                    nil,
+			DeprecatedFullNameOverride: "",
+			External:                   nil,
+			Force:                      nil,
+			FullnameOverride:           nil,
+			ImagePullSecrets:           nil,
+			LicenseKey:                 nil,
+			LicenseSecretRef:           nil,
+			Logging:                    nil,
+			Monitoring:                 nil,
+			NameOverride:               nil,
+			NodeSelector:               nil,
+			PostInstallJob:             nil,
+			PostUpgradeJob:             nil,
+			RBAC:                       nil,
+			RackAwareness:              nil,
+			Service:                    nil,
+			ServiceAccount:             nil,
+			Storage:                    nil,
+			Tests:                      nil,
+			Tolerations:                nil,
+			Tuning:                     nil,
+		}
+
+	case redpandav1alpha2.ClusterConfiguration:
+		return redpandav1alpha2.ClusterConfiguration(mapValues(src.Spec.ClusterConfig, convertFromValueSource))
+
+	case *redpandav1alpha2.Enterprise:
+		var ref *redpandav1alpha2.EnterpriseLicenseSecretRef
+		if value := src.Spec.Enterprise.License.ValueFrom; value != nil {
+			ref = &redpandav1alpha2.EnterpriseLicenseSecretRef{
+				Key:  ptr.To(value.SecretKeyRef.Key),
+				Name: ptr.To(value.SecretKeyRef.Name),
+			}
+		}
+		return &redpandav1alpha2.Enterprise{
+			License:          ptr.To(src.Spec.Enterprise.License.Value),
+			LicenseSecretRef: ref,
+		}
+
+	case *redpandav1alpha2.TLS:
+		certs := map[string]*redpandav1alpha2.Certificate{}
```

Comment:
how should we handle roundtrip conversions of people specifying randomly named certs? or should we just document that they aren't supported?

Reply:
So this case isn't present in v1alpha3 -> v1alpha2 -> v1alpha3 because the API doesn't permit it. I'll address it in  v1alpha2 -> v1alpha3 -> v1alpha2 conversion. My plan current plan is to:
- Shove the entire config into an annotation
- Rehydrate the annotation
- Convert into the rehydrated object

Doing so should preserve any fields that don't have an equivalent in the v1alpha3 API.